lua/damagelogs/client/init.lua

```diff
@@ -204,21 +204,21 @@ hook.Add("Think", "Think_Damagelog", function()
 end)
 
 function Damagelog:StrRole(role)
-    if not ROLES then
-        if role == ROLE_TRAITOR then
-            return TTTLogTranslate(GetDMGLogLang, "traitor")
-        elseif role == ROLE_DETECTIVE then
-            return TTTLogTranslate(GetDMGLogLang, "detective")
-        elseif role == DAMAGELOG_ROLE_DISCONNECTED then
-            return TTTLogTranslate(GetDMGLogLang, "disconnected")
-        else
-            return TTTLogTranslate(GetDMGLogLang, "innocent")
-        end
+    if role == DAMAGELOG_ROLE_DISCONNECTED then
+        return TTTLogTranslate(GetDMGLogLang, "disconnected")
     else
-        if role == DAMAGELOG_ROLE_DISCONNECTED then
-            return TTTLogTranslate(GetDMGLogLang, "disconnected")
-        else
+        if TTT2 then
             return TTTLogTranslate(GetDMGLogLang, GetRoleByIndex(role).name)
```

Comment:
Do we need to handle the situation where a TTT2 role is removed and `GetRoleByIndex(role)` returns nil?

Reply:
Hmm that wasn't handled previously and really shouldn't happen. `GetRoleByIndex(role)` is a built in TTT2 method that I believe catches that but I'd have to check on that.